### PR TITLE
Add support for variable env vars

### PR DESCRIPTION
**Pull request by LeoDiazL**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,7 +24,13 @@ You'll need [Access Keys](https://docs.aws.amazon.com/powershell/latest/userguid
 
 ## Environment variables
 
-For envirnoment variables in your app, you can provide a `repo_env` file in your repo, a `.env` file in GitHub Secrets named `DOT_ENV`, or an AWS Secret. Then hook it up in your `docker-compose.yaml` file like:
+For envirnoment variables in your app, you can provide
```
Comment (PhillypHenning) [inline README.md:27]: `provide:`
As it is denoting a list

File: operations/_scripts/generate/generate_app_repo.sh [hunk 2]
```diff
@@ -17,6 +17,9 @@ fi
 
 cp -rf "$TARGET_PATH"/* "${GITHUB_ACTION_PATH}/operations/deployment/ansible/app/${GITHUB_REPO_NAME}/"
 
-echo "Copying checked in env file from repo to Ansible deployment path"
-
-cp "$TARGET_PATH/$REPO_ENV" "${GITHUB_ACTION_PATH}/operations/deployment/ansible/repo.env"
\ No newline at end of file
+if [ -s "$TARGET_PATH/$REPO_ENV" ]; then
```
Comment (PhillypHenning) [inline operations/_scripts/generate/generate_app_repo.sh:20]: What's `-s` doing here out of curiousity?

File: operations/deployment/ansible/bitops.before-deploy.d/merge-tf-env.sh [hunk 3]
```diff
@@ -6,7 +6,7 @@ set -e
 echo "BitOps Ansible before script: Merge Terraform Enviornment Variables..."
 
 # Merging order
-order=tf,repo,ghs,aws
+order=tf,repo,ghv,ghs,aws
```
Comment (PhillypHenning) [inline operations/deployment/ansible/bitops.before-deploy.d/merge-tf-env.sh:9]: What is this?

File: operations/_scripts/generate/generate_app_repo.sh (hunk 2 above)
Comment (LeoDiazL) [reply operations/_scripts/generate/generate_app_repo.sh:20]: If it exists and not empty

File: action.yaml [hunk 4]
```diff
@@ -115,7 +118,8 @@ runs:
         TF_STATE_BUCKET: ${{ inputs.tf_state_bucket }}
         TF_STATE_BUCKET_DESTROY: ${{ inputs.tf_state_bucket_destroy }}
         REPO_ENV: ${{ inputs.repo_env }}
-        DOT_ENV: ${{ inputs.dot_env }}
+        GHS_ENV: ${{ inputs.dot_env }}
```
[the remark this answers is not in the document]
Comment (LeoDiazL) [reply action.yaml:121]: No, we need a proper input ... not the same :man_facepalming:

File: README.md (hunk 1 above)
Comment (LeoDiazL) [reply README.md:27]: Gotcha

File: operations/_scripts/generate/generate_dot_env.sh [hunk 5]
```diff
@@ -5,4 +5,5 @@ set -e
 
 echo "In generate_dot_env.sh"
 
-echo "$DOT_ENV" >> "${GITHUB_ACTION_PATH}/operations/deployment/ansible/ghs.env"
+echo "$GHV_ENV" > "${GITHUB_ACTION_PATH}/operations/deployment/ansible/ghv.env"
```
[the remark this answers is not in the document]
Comment (LeoDiazL) [reply operations/_scripts/generate/generate_dot_env.sh:8]: Nono ... 
GHS as in Github Secrets 
GHV as in Github Variables

File: operations/deployment/ansible/bitops.before-deploy.d/merge-tf-env.sh (hunk 3 above)
Comment (LeoDiazL) [reply operations/deployment/ansible/bitops.before-deploy.d/merge-tf-env.sh:9]: Github variables

File: action.yaml [hunk 6]
```diff
@@ -115,7 +118,8 @@ runs:
...
+        GHV_ENV: ${{ inputs.ghv_env }}
```
Comment (LeoDiazL) [inline action.yaml:122]: This is what's happening ...
DOT_ENV got renamed to GHS for Github Secrets
GHV is Github variables

File: action.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (LeoDiazL) [reply action.yaml:121]: Check last comment about lines 121 / 122

File: README.md [hunk 7]
```diff
@@ -24,7 +24,13 @@ You'll need [Access Keys](https://docs.aws.amazon.com/powershell/latest/userguid
 
 ## Environment variables
 
-For envirnoment variables in your app, you can provide a `repo_env` file in your repo, a `.env` file in GitHub Secrets named `DOT_ENV`, or an AWS Secret. Then hook it up in your `docker-compose.yaml` file like:
+For envirnoment variables in your app, you can provide:
+ - `repo_env` file in your repo
+ - `GH_vars` file in GitHub Variables
+ - `.env` file in GitHub Secrets
+ - `AWS Secret` in JSON format
```
Comment (mickmcgrath13) [inline README.md:31]: the syntax of the code snippets is inconsistent (e.g. `repo_env` vs `GH_vars` vs `AWS Secret`.  I think consistency would be good here.  how about
```suggestion
 - `repo_env` - A file in your repo that contains env vars
 - `ghv_env` An entry in GitHub Action Variables
 - `dot_env` An entry in GitHub Secrets
 - `aws_secret_env` The path to a JSON format secret in AWS
```

**Review by mickmcgrath13 — CHANGES_REQUESTED** [review]

**Review by mickmcgrath13 — APPROVED** [review]
seems legit

I didn't test, but the code lgtm

**Pull request merged**